includes/core/Contracts/API/Response/interface-base-response.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * Base response interface.
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Core\Contracts\API\Response;
+
+use WCPay\Core\DataTransferObjects\Response;
+use WP_Http_Cookie;
+
+/**
+ * Base response interface.
+ */
+interface Base_Response {
+
+	/**
+	 * Get body.
+	 *
+	 * @return string
+	 */
+	public function get_body();
+
+	/**
+	 * Get status code.
+	 *
+	 * @return int
+	 */
+	public function get_code();
+
+	/**
+	 * Get headers.
+	 *
+	 * @return array
+	 */
+	public function get_headers();
+
+	/**
+	 * Get message.
+	 *
+	 * @return string
+	 */
+	public function get_message();
+
+	/**
+	 * Get cookies.
+	 *
+	 * @return WP_Http_Cookie[]
+	 */
+	public function get_cookies();
+
+	/**
+	 * Get response body as array.
+	 *
+	 * @return array|mixed
+	 */
+	public function get_response_data();
+
+	/**
+	 * Determinate does error exists in response data.
+	 *
+	 * @return bool
+	 */
+	public function has_error_in_response_data();
+
+	/**
+	 * Create DTO from WCPay response.
+	 *
+	 * @param array $response Server response.
+	 * @return Response
+	 */
+	public static function create_from_wc_pay_response( $response);
```

Comment:
Already been mentioned in other comments, but I would like to see more explicit naming that contains "server". `wc_pay_response` is quite unclear within the context. It's WCpay plugin, so reading this, it feels as if we're doing something with the response we send to the frontend.

Reply:
I will update the function name to reflect from where the response is created.

Reply:
Fixed.